source/Gamerules.cpp

```diff
@@ -0,0 +1,78 @@
+/* Gamerules.cpp
+Copyright (c) 2022 by Amazinite
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "Gamerules.h"
+
+#include "DataNode.h"
+
+#include <algorithm>
+
+using namespace std;
+
+namespace {
+	// Returns true if "token" is a valid boolean.
+	bool ValidBool(const string &token) {
```

Comment:
Actually those could become useful on all tokens, so why dont we integrate them into the datanode class?